03/app.js

```diff
@@ -41,18 +40,58 @@ books.getAuthor = function(isbn) {
     return false;
 }
 
-books.getTitle = function(isbn, lang) {
+books.getTitle = function (isbn, lang) {
 
+    // check if input isbn exist in object
+    if (typeof this[isbn] === 'undefined') {
+        return null;
+    }
+
+    const title = this[isbn]['title'];
+
+    for (const key in title) {
+        if (key === lang) {
+            return title[key];
+        }
+    }
+
+    // check if input lang exist in object
+    if (typeof this[lang] === 'undefined') {
+        return null;
+    }
 }
 
-books.getTranslator = function(isbn, lang) {
+books.getTranslator = function (isbn, lang) {
+
+    // check if input isbn exist in object
+    if (typeof this[isbn] === 'undefined') {
+        return null;
+    }
+
+    const translator = this[isbn]['translator'];
+
+    for (const key in translator) {
+        if (lang === key) {
+
+            // check if property has null
+            if (translator[key] === null) {
+                return false;
+            }
+            return translator[key];
+        }
+    }
 
+    // check if input lang exist in object
+    if (typeof this[lang] === 'undefined') {
+        return null;
+    }
 }
 
 
-console.log( books.getAuthor('978-83-7278-000-3') ); // J.K. Rowling
-console.log( books.getAuthor('000-00-0000-000-0') ); // null
-console.log( books.getTitle('978-83-7278-000-3', 'pl') ); // Harry Potter i Kamień Filozoficzny
-console.log( books.getTitle('978-83-7278-000-3', 'en') ); // Harry Potter and the Philosopher's Stone
-console.log( books.getTranslator('83-7278-007-2', 'pl') ); // Andrzej Polkowski
-console.log( books.getTranslator('83-7278-007-2', 'en') ); // false
+
+console.log(books.getAuthor('978-83-7278-000-3')); // J.K. Rowling
```

Comment:
W tym zadaniu "kombinowanie" jest normalne, następnym razem będziesz bardziej świadomie to robił :)